services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -139,78 +141,117 @@ async function analyzeTable(table, config) {
+    primaryKeys.forEach((primaryKey) => {
+      if (primaryKey === constraint.columnName) {
+        const manyToManyKeys = _.filter(constraints, (otherKey) =>
+          otherKey.columnName !== constraint.columnName
+            && otherKey.columnType === FOREIGN_KEY
+            && primaryKeys.includes(otherKey.columnName)) || [];
+
+        manyToManyKeys.forEach((manyToManyKey) => {
+          addTableSchemaReference(
+            databaseSchema[referenceTableName],
+            referenceTableName,
+            ASSOCIATION_TYPE_BELONGS_TO_MANY,
+            constraint,
+            manyToManyKey,
+          );
+        });
+        isManyToMany = manyToManyKeys !== [];
+      }
+    });
+  }
+  if (!isManyToMany) {
+    addTableSchemaBelongsToIfReferenceIsPrimaryOrUnique(
+      databaseSchema[referenceTableName],
+      table,
+      constraint,
+    );
+    addTableSchemaReference(
+      databaseSchema[referenceTableName],
+      referenceTableName,
+      (isPrimary || isUnique) ? ASSOCIATION_TYPE_HAS_ONE : ASSOCIATION_TYPE_HAS_MANY,
+      constraint,
+    );
+  }
+}
+
+// NOTICE: Use the foreign key and reference properties to determine the associations and push them
+//         as references of the table.
+function defineAssociations(databaseSchema) {
   Object.values(databaseSchema).forEach((table) => {
     const { constraints, primaryKeys } = table;
     constraints.forEach((constraint) => {
-      const { columnName } = constraint;
-      const uniqueIndexes = constraint.uniqueIndexes || null;
-
-      if (constraint.columnType === 'FOREIGN KEY') {
-        const isInCompositeKey = primaryKeys
-          && primaryKeys.length > 1
-          && primaryKeys.includes(columnName);
-
-        const { isPrimary, isUnique } = checkUnicity(primaryKeys, uniqueIndexes, columnName);
-
-        const referenceTableName = constraint.foreignTableName;
-        const referenceColumnName = constraint.foreignColumnName;
-        let isManyToMany = false;
-
-        if (isInCompositeKey) {
-          // Check if the foreignKey is in a composite primary key
-          primaryKeys.forEach((primaryKey) => {
-            if (primaryKey === constraint.columnName) {
-              const manyToManyKeys = _.filter(constraints, (otherKey) =>
-                otherKey.columnName !== constraint.columnName
-                  && otherKey.columnType === 'FOREIGN KEY' && primaryKeys.includes(otherKey.columnName)) || [];
-
-              manyToManyKeys.forEach((manyToManyKey) => {
-                databaseSchema[referenceTableName].references.push(
-                  createReference(
-                    referenceTableName,
-                    ASSOCIATION_TYPE_BELONGS_TO_MANY,
-                    constraint,
-                    manyToManyKey,
-                  ),
-                );
-              });
-              isManyToMany = manyToManyKeys !== [];
-            }
-          });
-        }
-        if (!isManyToMany) {
```

Comment:
Isolated in `addTableSchemaBelongsToIfReferenceIsPrimaryOrUnique` function to reduce complexity.